compiler/main/driver.cpp

```diff
@@ -1142,6 +1166,8 @@ static ArgumentDescription arg_desc[] = {
  {"llvm", ' ', NULL, "[Don't] use the LLVM code generator", "N", &fYesLlvmCodegen, "CHPL_LLVM_CODEGEN", setLlvmCodegen},
  {"llvm-wide-opt", ' ', NULL, "Enable [disable] LLVM wide pointer optimizations", "N", &fLLVMWideOpt, "CHPL_LLVM_WIDE_OPTS", NULL},
  {"mllvm", ' ', "<flags>", "LLVM flags (can be specified multiple times)", "S", NULL, "CHPL_MLLVM", setLLVMFlags},
+ {"llvm-remarks", ' ', "<regex>", "Print LLVM optimization remarks", "S128", &llvmRemarksFilters, NULL, NULL},
+ {"llvm-remarks-function", ' ', "<name>", "Print LLVM optimization remarks only for these functions", "S", NULL, NULL, &setLLVMRemarksFunctions},
```

Comment:
Hmm, I wonder if `--llvm-print-ir` should be here, or if this new functionality should be next to `--llvm-print-ir`. I think `llvm-print-ir` is a Developer flag today.

Reply:
I chose to put this here instead of with `--llvm-print-ir` so that this can be used when not in developer mode. There is some filtering based on if it is developer mode or not. If we require users to do `--devel --llvm-remarks ...`, there is extra noise from internal/standard modules.

I think this could also be solved by adding another flag to control printing optimization remark from internal/standard modules.

Reply:
> If we require users to do `--devel --llvm-remarks ...`, there is extra noise from internal/standard modules.

Developer flags are available to everybody (developer mode or not). They just are listed separately and aren't documented in the man page.

IMO it would be reasonable to keep it as a Developer flag if you consider it experimental, which the technote seemed to suggest.

Reply:
That works for me, I'll move it around